Editor/AddressableImporter.cs

```diff
@@ -26,47 +26,81 @@ static void OnPostprocessAllAssets(string[] importedAssets, string[] deletedAsse
             Debug.LogWarningFormat("[AddressableImporter] import settings file not found.\nPlease go to Assets/AddressableAssetsData folder, right click in the project window and choose 'Create > Addressable Assets > Import Settings'.");
             return;
         }
-        else if (importSettings.rules == null || importSettings.rules.Count == 0)
+        if (importSettings.rules == null || importSettings.rules.Count == 0)
             return;
-        var entriesAdded = new List<AddressableAssetEntry>();
+
+        var dirty = false;
         var prefabStage = PrefabStageUtility.GetCurrentPrefabStage();
-        foreach (string assetPath in importedAssets)
+        foreach (var importedAsset in importedAssets)
         {
-            // Ignore current editing prefab asset.
-            if (prefabStage != null && prefabStage.prefabAssetPath == assetPath)
-                continue;
-            foreach (var rule in importSettings.rules)
+            if (prefabStage == null || prefabStage.prefabAssetPath != importedAsset) // Ignore current editing prefab asset.
+                dirty |= ApplyImportRule(importedAsset, settings, importSettings);
+
+            // Remove group if empty.
+            if (importSettings.removeEmtpyGroups && importedAsset.Contains("/AssetGroups/"))
```

Comment:
I can see this applies to the second import event caused by `AssetDatabase.SaveAssets`. But I don't get the benefits over the previous implementation.

```
if (importSettings.removeEmtpyGroups)
{
    settings.groups.RemoveAll(_ => _.entries.Count == 0 && !_.IsDefaultGroup());
}
```

Reply:
With the previous implementation, it seems that the folder is not deleted when the previous folder has been empty as a result of moving assets to a folder that does not match a rule.
I'd like to check this behavior again after the other discussions are over.

Reply:
I bet the behavior you described was caused by the previous implementation ignored the moved assets parameter, but not these lines.

Reply:
Fixed this at 8cb2602.
I used `settings.RemoveGroup` instead of `settings.groups.RemoveAll` and it worked fine.

Reply:
Looks good.